### PR TITLE
PaperClickPopover

### LINKED ISSUE #597: ClickPopover component
Want a popover which opens on a click rather than a hover!

Unfortunately the paperpopover uses the hoverpopover in https://github.com/openmsupply/openmsupply-client/tree/main/packages/common/src/ui/components/popover so won't do the job. 

Make `PaperPopover` .. `PaperHoverPopover` and make a `PaperClickPopover` 🤷 ? Or.. give `PaperPopover` a prop `opensOnHover`. My gut feeling is that prop alone basically changes most of the implementation of the component so just make the second one. But my other gut says that having two components is not a great api for a consumer. Life is so hard isn't it?

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #597 

Slightly overboard perhaps.. using inheritance and extending the current poppers etc. 
Renamed existing paper popover and added a click only version, and after than, wondering why I didn't just rename the hover and add an option 🙄

### CODE REVIEW

File: packages/common/src/ui/components/popover/ClickPopover/ClickPopover.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  hide,
+  Popover,
+}) => (
+  <>
+    <div style={{ cursor: 'pointer' }} onClick={show}>
+      {children}
+    </div>
+    <Popover placement={placement}>
+      <Box>
+        <IconButton
```
Comment (mark-prins) [inline packages/common/src/ui/components/popover/ClickPopover/ClickPopover.tsx:28]: ok, so maybe worth it. added a close button to the click popover, which isn't required on the hover version.

File: packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/PaperClickPopover.tsx [hunk 2]
```diff
@@ -39,20 +41,6 @@ export const PaperPopover: FC<PaperPopoverProps> = ({
       }
     >
       {children}
-    </HoverPopover>
+    </ClickPopover>
   );
 };
-
-export interface PaperPopoverSectionProps {
```
Comment (mark-prins) [inline packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/PaperClickPopover.tsx:46]: moved these out so I could reuse them

File: packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/usePaperClickPopover.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React, { FC } from 'react';
+import { usePopover } from '../../BasePopover';
+import {
+  PaperClickPopover as ClickPopover,
+  PaperClickPopoverProps,
+} from './PaperClickPopover';
+
+export const usePaperClickPopover = () => {
```
Comment (mark-prins) [inline packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/usePaperClickPopover.tsx:8]: made the hook to make usage easier. My problem was wanting to use `hide` in both the content and in the popover - so I lifted that out to the consumer, which meant passing it back in, which is annoying.

Comment (joshxg) [reply packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/usePaperClickPopover.tsx:8]: Ah yes I see the problem.. annoying! Good work around though, makes it nice and easy

File: packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/PaperClickPopover.tsx (hunk 2 above)
Comment (joshxg) [reply packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/PaperClickPopover.tsx:46]: very nice

File: packages/common/src/ui/components/popover/ClickPopover/ClickPopover.tsx (hunk 1 above)
Comment (joshxg) [reply packages/common/src/ui/components/popover/ClickPopover/ClickPopover.tsx:28]: Ah yes.

I like the close button but I also wonder if clicking away should close also.. that was my expectation.. or, at least when navigating away, I suppose.

I had a play and adding `<ClickAwayListener onClickAway={hide}>` around the `Box` seems to work fine if you think so also

File: packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/usePaperClickPopover.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  const { show, hide, Popover } = usePopover();
+  const PaperClickPopover: FC<
+    Omit<PaperClickPopoverProps, 'show' | 'hide' | 'Popover'>
+  > = props => (
+    <ClickPopover show={show} hide={hide} Popover={Popover} {...props} />
+  );
+
+  return { show, hide, PaperClickPopover };
```
Comment (joshxg) [inline packages/common/src/ui/components/popover/PaperPopover/PaperClickPopover/usePaperClickPopover.tsx:16]: very nice!

**Review by joshxg — APPROVED** [review]
Love it!

**Review by joshxg — APPROVED** [review]
Ah .. I noticed there was a slight delay in opening / closing which was irking me .. I thought it might have been a render issue, but it's actually that the `Popover` is using `useDebounceCallback` (was for the hovering 🙄 ) with a delay of 250ms on opening and 500ms on closing..

Adding some optional parameters for the timeout to `usePopover` should be a straight forward change to get rid of the delay

**Comment by joshxg** [comment]
My suggestions are minor and could be ignored.. since I'm shellfish and want to utilise the store picker.. lets get it in 😁

File: packages/common/src/ui/components/popover/ClickPopover/ClickPopover.tsx (hunk 1 above)
Comment (mark-prins) [reply packages/common/src/ui/components/popover/ClickPopover/ClickPopover.tsx:28]: yeah - was undecided, my initial reaction was that click away should work. thanks for adding it in

**Pull request merged**